src/openrct2/object/Object.h

```diff
@@ -126,6 +128,46 @@ struct rct_object_entry_group
 assert_struct_size(rct_object_entry_group, 8);
 #endif
 
+enum class ObjectGeneration : uint8_t
+{
+    DAT,
+    JSON,
+};
+
+struct ObjectEntryDescriptor
+{
+    ObjectGeneration Generation;
+    union
+    {
+        rct_object_entry Entry; // For DAT objects
+        char Identifier[64];    // For JSON objects
+    };
+
+    ObjectEntryDescriptor() = default;
+    explicit ObjectEntryDescriptor(const rct_object_entry& newEntry)
+    {
+        Generation = ObjectGeneration::DAT;
+        Entry = newEntry;
+    }
+
+    explicit ObjectEntryDescriptor(std::string_view newIdentifier)
+    {
+        Generation = ObjectGeneration::JSON;
+        safe_strcpy(const_cast<char*>(Identifier), std::string(newIdentifier).c_str(), 64);
+    }
+
```

Comment:
Add a GetIdentifier function as its used multiple times.

Reply:
I don't think I understand. This function is setting stuff, not getting.

Reply:
There are a two locations (so far) where you have to get either the identifier or the dat name. Just make a function called GetIdentifier and use that for those two locations.

Reply:
No, it is either the JSON identifier (a string) or a DAT _header_ (which is a struct and not a string). And even if I could create such a function, there is still the issue that I have to _set_ the property based on whether we're handling a DAT or JSON object.

Reply:
You have two instances where you branch to get the identifier or the dat name. FindObject and window_editor_object_selection_select_object use a getter to get the identifier name.

Reply:
Right, that makes sense. I was confused because you made this a line note.